src/Interpreters/Aggregator.cpp

```diff
@@ -2601,6 +2601,21 @@ void NO_INLINE Aggregator::mergeWithoutKeyDataImpl(
 
     AggregatedDataVariantsPtr & res = non_empty_data[0];
 
+    for (size_t i = 0; i < params.aggregates_size; ++i)
+    {
+        if (aggregate_functions[i]->isAbleToParallelizeMerge() &&
```

Comment:
`isAbleToParallelizeMerge` looks irrelevant here, `isParallelizeMergePrepareNeeded` should be enough

Reply:
Keep the `isAbleToParallelizeMerge` and delete 'isParallelizeMergePrepareNeeded'.

Reply:
The flag `isAbleToParallelizeMerge` is for the parallel merge of UniqExact. If the merge stage is not parallel (`isAbleToParallelizeMerge = false`), we could still do hashset converting in parallel. The merge stage and hashset converting stage are independent and irrelevant. Then I think `isAbleToParallelizeMerge` could be removed.